cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/distributed/DistributedProgramRunner.java

```diff
@@ -366,6 +368,20 @@ public ProgramController call() throws Exception {
     }
   }
 
+  private void addAdditionalLogAppenderJars(CConfiguration cConf, File tempDir,
+                                            Map<String, LocalizeResource> localizeResources) throws IOException {
+    if (!Strings.isNullOrEmpty(cConf.get(Constants.Logging.LOG_APPENDER_PROVIDER))) {
+      String provider = cConf.get(Constants.Logging.LOG_APPENDER_PROVIDER);
+      String jarDir = cConf.get(Constants.Logging.LOG_APPENDER_EXT_DIR);
+      File bundleJarFile = new File(tempDir, "log-appender.jar");
+      BundleJarUtil.createJar(new File(jarDir + "/" + provider), bundleJarFile);
```

Comment:
Why if the directory doesn't exist? This will throw IOException and fail the run?

Reply:
If that's the case, we should have a startup check for this config.

Reply:
Similar to other usages of this method, if the provided directory does not exist, IOException will be thrown.